go-apps/meep-rnis/server/rnis.go

```diff
@@ -354,29 +343,37 @@ func checkForExpiredSubscriptions() {
 		if expiryTime <= nowTime {
 			subscriptionExpiryMap[expiryTime] = nil
 			for _, subsId := range subsIndexList {
+				cbRef := ""
 				if ccSubscriptionMap[subsId] != nil {
+					cbRef = ccSubscriptionMap[subsId].CallbackReference
```

Comment:
Can "subsId" be the same for different types? If so then you need to know the subscription type in order to retrieve the correct callback reference. Otherwise you may find a match in the wrong subscription type map.

Reply:
actually, each subId is unique and cannot be part of many different types... we have many maps and we use a map just when necessary... but then when it is type to expire or delete, we are not aware which subscription type it is... which is why we g through all the maps...